src/objects/list/list.scss

```diff
@@ -0,0 +1,56 @@
+@use '../../design-tokens/colors.yml';
+@use '../../design-tokens/sizes.yml';
+
+/**
+ * Override browser defaults. Note that this means it's very important to
+ * include `[role="list"]` to prevent disruption of list navigation in Safari
+ * VoiceOver.
+ *
+ * @see https://www.scottohara.me/blog/2019/01/12/lists-and-safari.html
+ */
+
+.o-list {
+  list-style: none;
+  padding-left: 0;
+}
+
+/**
+ * Modifier: Inline
+ *
+ * By default, this uses negative margins on the sides to account for extra
+ * margin between child elements. This allows child elements to reach the edges
+ * of their parent while maintaining consistent gaps between.
+ *
+ * This would be a _lot_ more elegant if it used the Flexbox-compatible `gap`
+ * property. But unfortunately there is no way to test for that without false
+ * positives as of this writing.
+ *
+ * @see https://medium.com/@schofeld/mind-the-flex-gap-c9cd1b4b35d8
+ * @see https://github.com/w3c/csswg-drafts/issues/3559
+ */
+
+.o-list--inline {
+  display: flex;
+  flex-wrap: wrap;
+  margin-left: sizes.$list-inline-gap / -2;
+  margin-right: sizes.$list-inline-gap / -2;
+}
+
+.o-list--inline > * {
+  margin-left: sizes.$list-inline-gap / 2;
+  margin-right: sizes.$list-inline-gap / 2;
+}
+
+/**
+ * Modifier: Ruled
+ */
+
+.o-list--ruled > * + * {
```

Comment:
Perhaps a silly question, but do we care about what happens if a user applies both `o-list--inline` and `o-list--ruled`?

Reply:
I asked myself the same thing. Right now they are completely incompatible.

I considered adding a `:not` selector or even a combined selector that inserts vertical rules between items, but since there isn't a place where we're doing that on the current site, it felt like overkill.

I'm open to suggestions!

Reply:
I think it's totally find to just add a note on the stories page saying "These two modifiers don't play nice with each other."

We don't need to cover every possible use case, but an acknowledgement that some combinations aren't supported is probably worth doing.

Reply:
I agree 💯 with what @spaceninja said here. 😄